### DISCUSSION TITLE
Use One Supabase Instance as the Authentication Provider for Other Supabase Instances?

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#16489] [Feature Requests]
I was wondering if anyone has ever attempted or successfully been able to use one supabase instance as an authentication provider (in the same way as Keycloak, Google, Facebook, etc) for other supabase instances? 

Here's the use case: 
- we have some clients who use a shared/multi-tenant database environment
- we have some clients who require a single-tenant database environment
- we have some clients who will be using both (i.e., one of their corporate accounts pays the premium for single tenancy and the other does not)
- we want to have a centralized, unified login experience (i.e., a single page that shows all of a user's accounts, one per company/tenant)

# Current Solution
- we're running a Keycloak instance to use as a central authentication provider
- the issue is that this requires us to federate the supabase instances, such that the central keycloak server needs its own supabase database to pool/collect data from all the federated servers/supabase instances (i.e., stream all the data from the isolated servers into one, centralized federation supabase project, effectively creating the Supabase authentication provider we need, but with the added headache of configuring keycloak and needing to network between keycloak and the central supabase instance)

# What We're Looking For
- basically, in the same way that we can use Keycloak or Facebook or Google as an authentication provider for a supabase instance, we want to use one **supabase** instance as the authentication provider for another supabase instance

If this is not something that's possible, perhaps I make this a feature request?
